usr/src/cmd/fs.d/df.c

```diff
@@ -1282,15 +1283,33 @@ adjust_total_blocks(struct df_request *dfrp, fsblkcnt64_t *total,
 		}
 
 		this_quota = _zfs_prop_get_int(zhp, ZFS_PROP_QUOTA);
-		if (this_quota && this_quota < quota)
+		if (this_quota && this_quota < quota) {
 			quota = this_quota;
+			logical = FALSE;
+		}
+
+		this_quota = _zfs_prop_get_int(zhp, ZFS_PROP_LOGICALREFQUOTA);
+		if (this_quota && this_quota < quota) {
```

Comment:
I'm not super familiar with this `df` code, but I think that assuming the current code makes sense, the logical (ha ha) extension would be to check the `LOGICALREFQUOTA` only `if (first)`, as we do for `REFQUOTA`.